### PR TITLE
Add concourse terraform-resource

### PR DESCRIPTION
**Pull request by duboisph**
# Description

We're using the TF kubernetes and helm providers using the exec method, which in our case needs awscli.

# Related Issue

https://github.com/skyscrapers/engineering/issues/780

# Motivation and Context

# How Has This Been Tested?

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] I have followed the PR process as described [here](https://github.com/skyscrapers/documentation/blob/master/coding_guidelines/git.md#pull-requests)
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: terraform-resource/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# https://github.com/alpine-docker/k8s/blob/master/Dockerfile
+RUN apk add --update --no-cache python3 && \
+  python3 -m ensurepip && \
+  pip3 install --upgrade pip && \
+  pip3 install awscli && \
+  pip3 cache purge
+
+# Install aws-iam-authenticator
+# https://github.com/alpine-docker/k8s/blob/master/Dockerfile
+RUN authenticator=$(aws --no-sign-request s3 ls s3://amazon-eks --recursive |grep aws-iam-authenticator$|grep amd64 |awk '{print $NF}' |sort -V|tail -1) && \
```
Comment (iuriaranda) [inline terraform-resource/Dockerfile]: :thinking: is this really needed? I thought that the awscli could authenticate to k8s by its own nowadays.

Comment (duboisph) [reply terraform-resource/Dockerfile]: Not 100% sure, but I think that aws only the case for awscli v2, while we'rer installing v1 here (*).


Installing v2 on alpine is quite complex, as it's build on glibc, so need to pull in compatibility layer etc.

Comment (duboisph) [reply terraform-resource/Dockerfile]: Ok, not needed after all, removing

Comment (iuriaranda) [reply terraform-resource/Dockerfile]: hmm I see, didn't notice it was v1. Tbh I don't think it's such a good idea to install and use an already deprecated version of a tool. Shouldn't be that hard to install v2 afaik, just as it's done in the official image: https://github.com/aws/aws-cli/blob/v2/docker/Dockerfile no?

Comment (duboisph) [reply terraform-resource/Dockerfile]: No, it's harder than that. The official image uses amazonlinux, while this image is alpine based. Alpine doesn't come with glibc, which the awscliv2 is compiled against. The binary won't run.

It's possible via `gcompat`, but that might maybe cause other issues down the road https://wiki.alpinelinux.org/wiki/Running_glibc_programs

**Review by iuriaranda — APPROVED** [review]

**Pull request merged**